tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -0,0 +1,202 @@
+        private static readonly ConcurrentDictionary<string, MetricPoint> CapturedMetrics = new();
+
+        private static readonly Dictionary<string, object?> EmptyTags = new();
+        private static string? _cachedTemporality;
+
+        public static void OnInstrumentPublished(Instrument instrument, MeterListener listener)
+        {
+            bool shouldEnable;
+            var meterName = instrument.Meter.Name;
+            var enabledMeterNames = Tracer.Instance.Settings.OpenTelemetryMeterNames;
+
+            if (enabledMeterNames.Length > 0)
+            {
+                shouldEnable = enabledMeterNames.Contains(meterName);
+            }
+            else
+            {
+                shouldEnable = !meterName.StartsWith("System.", StringComparison.Ordinal);
+            }
+
+            if (shouldEnable)
+            {
+                listener.EnableMeasurementEvents(instrument, state: null);
+                Log.Debug("Enabled measurement events for instrument: {InstrumentName} from meter: {MeterName}", instrument.Name, meterName);
+            }
+        }
+
+        public static void OnMeasurementRecordedLong(Instrument instrument, long value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            ProcessMeasurement(instrument, value, tags, isInteger: true);
+        }
+
+        public static void OnMeasurementRecordedDouble(Instrument instrument, double value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            ProcessMeasurement(instrument, value, tags, isInteger: false);
+        }
+
+        private static void ProcessMeasurement<T>(Instrument instrument, T value, ReadOnlySpan<KeyValuePair<string, object?>> tags, bool isInteger)
+            where T : struct
+        {
+            try
+            {
+                var instrumentType = instrument.GetType().FullName;
+                if (instrumentType is null)
+                {
+                    Log.Warning("Unable to get the full name of the instrument type for: {InstrumentName}", instrument.Name);
+                    return;
+                }
+
+                ProcessMeasurementCore(instrument, instrumentType, value, tags, isInteger);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error processing measurement for instrument: {InstrumentName}", instrument.Name);
+            }
+        }
+
+        private static void ProcessMeasurementCore<T>(Instrument instrument, string instrumentType, T value, ReadOnlySpan<KeyValuePair<string, object?>> tags, bool isInteger)
+            where T : struct
+        {
+            string aggregationType;
+            if (instrumentType.StartsWith("System.Diagnostics.Metrics.Counter`1") ||
+                instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableCounter`1"))
+            {
+                aggregationType = "Counter";
```

Comment:
Small change: You should use an enum rather than arbitrary strings to keep track of the instrument type. This can also help you determine if you should be recording `long` values vs `double` values, which I think you should implement. Converting `long` values into `double` values is adding overhead that I think we can avoid